app/views/work/edit.html.slim

```diff
@@ -239,3 +242,30 @@
         };
       });
     });
+
+-content_for :javascript
+  javascript:
+    $(document).ready(function() {
+      const collectionSelect = $('#work_collection_id');
+      const documentSetsDiv = $('#document_sets');
+
+      function updateDocumentSetOptions(selectedId) {
+        $.ajax({
+          url: "#{work_document_sets_select_path}",
+          method: 'GET',
+          data: { collection_id: selectedId },
+          success: function(response) {
+            documentSetsDiv.html(response);
+            $('#work_document_set_ids').multiSelect();
+          },
+          error: function(xhr, status, error) {
+            console.error('Error fetching document sets:', error);
+          }
+        });
+      }
+
+      collectionSelect.on('change', function() {
+        const selectedId = $(this).val();
+        updateDocumentSetOptions(selectedId);
```

Comment:
On collection select change, we pass in the selected id which we will be using as params[:collection_id] for the controller